### ISSUE TITLE
ssd_resnet50_v1_fpn fine tune 

### ISSUE THREAD

**Issue by Ajders1** [tensorflow/models#8391]
Hi,

I want to train ssd_resnet50_v1 on my own dataset locally.

OS: Windows 10
Python: 3.6.10
TF: 1.12.0 GPU version from pip
GPU: Nvidia GTX 1080 Ti

I try to train through **main_model.py** and **train.py** (legacy). The command which I use is:

> python object_detection/train.py --train_dir=object_detection/train --pipeline_config_path=object_detection/ownTraining/Pipelinessd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03.config

With both commands, I get a lot of warnings like this (there is only a few):

> WARNING:root:Variable [resnet_v1_50/block1/unit_1/bottleneck_v1/conv1/BatchNorm/beta] is not available in checkpoint
> WARNING:root:Variable [resnet_v1_50/block1/unit_1/bottleneck_v1/conv1/BatchNorm/gamma] is not available in checkpoint
> WARNING:root:Variable [resnet_v1_50/block1/unit_1/bottleneck_v1/conv1/BatchNorm/moving_mean] is not available in checkpoint
> WARNING:root:Variable [resnet_v1_50/block1/unit_1/bottleneck_v1/conv1/BatchNorm/moving_variance] is not available in checkpoint
> WARNING:root:Variable [resnet_v1_50/block1/unit_1/bottleneck_v1/conv1/weights] is not available in checkpoint

I first start with TF 2.1.0. After that, I reinstall TF to 1.15.0. With TF 1.15.0 was not possible to start training, because other errors occurred. With TF 1.12.0 I´m able to train model, but with these warnings. 
I found that problem in some other issues, but neither has solved this problem.

Could someone help me?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Ajders1** [comment]
**OS:** Windows 10
**Python:** 3.6.10
**TF:** 1.12.0 GPU version from pip
**GPU:** Nvidia GTX 1080 Ti 11GB
**Bazel:** 0.16.1 (not working now)
**CUDA:** 9.0
**cudnn64_7.dll** (looks that pip downloaded separately and don't use my downloaded file)

**Top-level directory:** D:\models\research\

> python object_detection/train.py --train_dir=object_detection/train --pipeline_config_path=object_detection/ownTraining/Pipelinessd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03.config

**Comment by VenuGopalVasarla** [comment]
I too got the same issue with other resnet models while trying to fine tune.  Please help me to solve this.

**Comment by Ajders1** [comment]
I fixed it as s

**Comment by Ajders1** [comment]
I use configs from samples (https://github.com/tensorflow/models/tree/master/research/object_detection/samples/configs) and fixed it by adding **from_detection_checkpoint: true** under **fine_tune_checkpoint:**

I found that solution here: https://github.com/tensorflow/models/issues/4940

**Comment by shua1zhang** [comment]
> I use configs from samples (https://github.com/tensorflow/models/tree/master/research/object_detection/samples/configs) and fixed it by adding **from_detection_checkpoint: true** under **fine_tune_checkpoint:**
> 
> I found that solution here: #4940

Have you successfully load the pretrained model and how is mmAP at the beginning of finetuning? 
I have the same issue and could solve the error by adding that line. However the mmAP is pretty low and I'm afraid that maybe ckpt is not correctly loaded?

**Comment by Ajders1** [comment]
mAP starts at low values and then it grows, but I use own dataset, so it should be right. When I train YOLO (https://github.com/AlexeyAB/darknet) with the same dataset, mAP also starts at low values, and then it grows.
There was no indication that they were not be loaded. I didn't compare with and without fine-tune training. Did you compare it?

**Comment by shua1zhang** [comment]
> mAP starts at low values and then it grows, but I use own dataset, so it should be right. When I train YOLO (https://github.com/AlexeyAB/darknet) with the same dataset, mAP also starts at low values, and then it grows.
> There was no indication that they were not be loaded. I didn't compare with and without fine-tune training. Did you compare it?

Finally after I modify original pipeline.config file copied from sameple/config folder and add following two lines, I was able to correctly load pretrained resnet50_fpn_ssd model and get correct mmAP.

  from_detection_checkpoint: True
  load_all_detection_checkpoint_vars: true

@pkulzc shall we add these two lines into example pipeline.config files? 
Thank you.
